File: tools/test_golden.php
<?php
require_once __DIR__ . '/../src/php/FluxConstraint.php';

$vectors = json_decode(file_get_contents(__DIR__ . '/golden_vectors.json'), true);
$mismatches = 0;

foreach ($vectors as $v) {
    $cs = array_map(fn($c) => ['lo' => $c['lo'], 'hi' => $c['hi'], 'name' => ''], $v['constraints']);
    $fc = new FluxConstraint($cs);
    $r = $fc->check($v['value']);
    $exp = $v['expected'];
    if ($r->error_mask !== $exp['error_mask'] || $r->isPass() !== $exp['passed']) {
        $mismatches++;
        if ($mismatches <= 5) echo "MISMATCH #{$v['id']}: value={$v['value']} got mask={$r->error_mask} expected={$exp['error_mask']}\n";
    }
}

echo "\nPHP: " . count($vectors) . " vectors, $mismatches mismatches\n";
exit($mismatches > 0 ? 1 : 0);

File: tools/test_golden.rb
require 'json'
require_relative '../src/ruby/flux_constraint'

vectors = JSON.parse(File.read(File.join(__dir__, 'golden_vectors.json')))
mismatches = 0

vectors.each do |v|
  rules = v['constraints'].map { |c| Flux::FluxRule.new("c", c['lo'], c['hi'], Flux::CAUTION, Flux::WARNING, Flux::CRITICAL) }
  fc = Flux::ConstraintChecker.new(rules)
  r = fc.check(v['value'])
  exp = v['expected']
  # Ruby FluxResult: Struct with passed, severity, violations
  if r.passed != exp['passed']
    mismatches += 1
    puts "MISMATCH ##{v['id']}: value=#{v['value']} got passed=#{r.passed} expected=#{exp['passed']}" if mismatches <= 5
  end
end

puts "\nRuby: #{vectors.length} vectors, #{mismatches} mismatches"
exit(mismatches > 0 ? 1 : 0)
